code/modules/vending/_vending.dm

```diff
@@ -646,11 +661,18 @@ GLOBAL_LIST_EMPTY(vending_products)
 	. = TRUE
 	if(!user.transferItemToLoc(I, src))
 		return FALSE
+	to_chat(user, span_notice("You insert [I] into [src]'s input compartment."))
+
+	for(var/datum/data/vending_product/product_datum in product_records + coin_records + hidden_records)
+		if(istype(I, product_datum.product_path))
```

Comment:
Might be safer to use `ispath()` isntead of `istype()`, unless you specifically want people to return subtypes as the normal product